SFLoggingKit/Classes/SFLogWriter.swift

```diff
@@ -0,0 +1,385 @@
+//
+//  SFLogWriter.swift
+//  SFLoggingKit
+//
+//  Created by Aleksandar Sergeev Petrov on 7.07.20.
+//
+
+import Foundation
+import os.log
+
+// NOTE: Writers don't append new line automatically.
+// If proper add it in log message string
+
+/// The `SFLogWriter` protocol defines a single API for writing a log message. The message can be written in any way the conforming object sees fit.
+/// For example, it could write to the console, write to a file, remote log to a third party service, etc.
+public protocol SFLogWriter {
+
+    func log(_ message: String, logLevel: SFLogLevel)
+}
+
+// MARK: - Xcode Console
+
+/// The SFConsoleLogWriter class runs all modifiers in the order they were created and prints the resulting message  to the Xcode console.
+open class SFConsoleLogWriter: SFLogWriter {
+
+    // MARK: - Properties
+
+    /// Determines if the logs should be visible when attatching a device and debugging in the console
+    private let shouldLogInBackgroundConsole: Bool
+
+    // MARK: - Initializers
+
+    /// Initializes a console writer instance.
+    ///
+    /// - Parameter shouldLogInBackgroundConsole: The function to use when logging to the console. Defaults to `print`.
+    ///
+    /// - Returns: A new console writer instance.
+    public init(shouldLogInBackgroundConsole: Bool = false) {
+        self.shouldLogInBackgroundConsole = shouldLogInBackgroundConsole
+    }
+
+    // MARK: - SFLogWriter
+
+    /// Writes the message to the console using the global `print` or `NSLog`  function.
+    ///
+    /// Modifier is run over the message in the order to provide log level before writing the message to
+    /// the console.
+    ///
+    /// - Parameters:
+    ///   - message: The original message to write to the console.
+    ///   - logLevel: The log level associated with the message.
+    open func log(_ message: String, logLevel: SFLogLevel) {
+        if shouldLogInBackgroundConsole {
+            NSLog(message)
+        } else {
+            print(message)
+        }
+    }
+
+}
+
+// MARK: - OSLog
+
+/// The `SFSystemLogWriter` class runs all modifiers in the order they were created and passes the resulting message
+/// off to an `OSLog` with the specified subsystem and category.
+/// NOTE: In iOS 14.0 Apple introduced `Logger` which does exactly the same
+open class SFSystemLogWriter: SFLogWriter {
+
+    // MARK: - Properties
+
+    public let subsystem: String
+    public let category: String
+
+    // MARK: - Initializers
+
+    /// Creates an `SFSystemLogWriter` instance from the specified `subsystem` and `category`.
+    ///
+    /// - Parameters:
+    ///   - subsystem: The subsystem. Default is app bundle identifier
+    ///   - category: The category. E.g. `ui`, `firebase`, `networking`, etc.
+    ///
+    /// - Returns: A new `SFSystemLogWriter` instance.
+    public init(subsystem: String = Bundle.main.bundleIdentifier!, category: String) {
```

Comment:
[optional]

Could we avoid this force unwrap?

Reply:
good catch